stan/math/fwd/scal/fun/is_nan.hpp

```diff
@@ -2,21 +2,23 @@
 #define STAN_MATH_FWD_SCAL_FUN_IS_NAN_HPP
 
 #include <stan/math/fwd/core.hpp>
+#include <stan/math/fwd/scal/meta/is_fvar.hpp>
 #include <stan/math/prim/scal/fun/is_nan.hpp>
 
 namespace stan {
 namespace math {
 
 /**
- * Returns 1 if the input's value is NaN and 0 otherwise.
+ * Returns true if the input's value is NaN and false otherwise.
  *
  * Delegates to <code>is_nan</code>.
  *
  * @param x Value to test.
- * @return <code>1</code> if the value is NaN and <code>0</code> otherwise.
+ * @return <code>true</code> if the value is NaN and <code>false</code>
+ * otherwise.
  */
 template <typename T>
-inline int is_nan(const fvar<T>& x) {
+inline typename std::enable_if_t<is_fvar<T>::value, bool> is_nan(const T& x) {
```

Comment:
What was the motivation for switching to `enable_if_t` from the simple pattern matching?  I think this signature should just be:

```
template <typename T>
inline bool is_nan(const fvar<T>& x);
```

Reply:
This was one was based on @syclik's review:
> Hmm... this looks odd. In develop, we have 3 definitions of is_nan: prim/scal, rev/scal, fwd/scal. They are implemented using function overloading.

> I think you either want to go the same route or you want to make this function the template function (as you have it) and then change the implementations in rev and fwd to be template specializations.